File: 09-leap-check/leap_check.php
<?php
// Leap check program in PHP
// Written by M.V.Harish Kumar on 17-09-2024

function is_leap($yr) {
	return ($yr % 4 == 0) && ($yr % 100 != 0) || ($yr % 400 == 0);
}

echo "Leap year checker\n";
$year = intval(readline('Enter a year: '));
if (is_leap($year)) {
	echo "$year is a Leap Year\n";
} else {
	echo "$year is NOT a Leap Year\n";
}

File: 09-leap-check/leap_check.rb
# Ruby program to check if given year is leap
# Written by M.V.Harish Kumar on 26-01-2025

def is_leap?(yr)
  (yr % 4 == 0) and (yr % 100 != 0) or (yr % 400 == 0)
end

puts "Leap year checker"
print "Enter the year: "
yr = gets.to_i
puts "#{yr} is #{'NOT ' unless is_leap? yr}a Leap Year"
